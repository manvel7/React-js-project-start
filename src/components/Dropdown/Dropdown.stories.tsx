import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Dropdown, {IProps} from "./Dropdown"
import DropdownItem from "./DropdownMenuItem"

const meta: Meta<IProps> = {
  title: "Components/Dropdown",
  component: Dropdown
}

const DropdownStory: Story<IProps> = (args) => (
  <Dropdown {...args}>
    <DropdownItem>Item 1</DropdownItem>
    <DropdownItem>Item 2</DropdownItem>
  </Dropdown>
)

const Medium = DropdownStory.bind({})
Medium.args = {
  size: "md",
  title: "Action"
}

const Small = DropdownStory.bind({})
Small.args = {
  size: "sm",
  title: "Action"
}

export {Medium, Small}

export default meta
