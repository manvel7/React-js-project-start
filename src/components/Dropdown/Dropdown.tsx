import React, {
  FunctionComponent,
  HTMLAttributes,
  MouseEventHandler,
  TransitionEvent,
  useEffect,
  useState
} from "react"
import {ButtonSize} from "../Button/Button"
import DropdownButton from "./Button"
import DropdownMenu from "./Menu"
import DropdownWrapper from "./Wrapper"

export type DisplayType = "block" | "none"

export interface IProps extends HTMLAttributes<HTMLDivElement> {
  size?: ButtonSize
  title: string
  onButtonClick?: MouseEventHandler<HTMLButtonElement>
  onToggleClick?: () => void
}

const Dropdown: FunctionComponent<IProps> = (props) => {
  const [isOpened, setIsOpened] = useState<boolean>(false)
  const [display, setDisplay] = useState<DisplayType>("none")
  const {children, size, title, onButtonClick, onToggleClick} = props

  const open = async () => {
    if (onToggleClick) {
      await onToggleClick()
    }

    setDisplay("block")
  }

  const onToggleOpen = () => {
    if (isOpened) {
      setIsOpened(false)
    } else {
      open()
    }
  }

  const onTransitionEnd = (event: TransitionEvent<HTMLDivElement>) => {
    if (event.propertyName === "opacity" && !isOpened) {
      setDisplay("none")
    }
  }

  useEffect(() => {
    if (display === "block") {
      setIsOpened(true)
    }
  }, [display])

  return (
    <DropdownWrapper>
      <DropdownButton
        isOpened={isOpened}
        onToggleOpen={onToggleOpen}
        title={title}
        size={size}
        onButtonClick={onButtonClick}
      />
      <DropdownMenu
        isOpened={isOpened}
        displayValue={display}
        onTransitionEnd={onTransitionEnd}>
        {children}
      </DropdownMenu>
    </DropdownWrapper>
  )
}

export default Dropdown
