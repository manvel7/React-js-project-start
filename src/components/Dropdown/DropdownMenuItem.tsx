import styled from "styled-components"

const DropdownMenuItem = styled.a`
  background-color: ${({theme}) => theme.palette.background.light};
  border: 0.1rem solid ${({theme}) => theme.palette.border};
  border-radius: 0.4rem;
  cursor: pointer;
  display: block;
  padding: 0.8rem 1.6rem;
  transition: background-color 0.15s ease-in;
  width: 100%;

  &:hover {
    background-color: ${({theme}) => theme.palette.background.tinted};
  }
`

export default DropdownMenuItem
