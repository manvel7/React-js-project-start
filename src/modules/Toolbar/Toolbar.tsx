import styled from "styled-components"

const Toolbar = styled.div`
  align-items: center;
  border-bottom: 0.1rem solid ${({theme}) => theme.palette.border};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  /* TODO: check this to match Navbar height */
  padding: 1.55rem;

  & > div > div {
    margin-right: 1.6rem;
  }

  .return-button {
    margin-right: 1.6rem;

    @media (min-width: ${({theme}) => theme.grid.breakpoints.sm}) {
      visibility: hidden;
    }
  }
`

export default Toolbar
