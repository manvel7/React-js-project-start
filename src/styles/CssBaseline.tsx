import {
  createGlobalStyle,
  css,
  DefaultTheme,
  GlobalStyleComponent,
  StyledProps
} from "styled-components"
import {mediaBreakpointDown} from "../utils/breakpoint"
import {getValue, toPercent} from "../utils/unit"
import {
  bodyMixin,
  smallMixin,
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin
} from "./typography"

interface IProps {}

const genericResetMixin = css<StyledProps<IProps>>`
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }

  ol,
  ul {
    list-style: none;
  }

  blockquote,
  q {
    quotes: none;
  }

  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
`

const appResetMixin = css<StyledProps<IProps>>`
  html {
    /* stylelint-disable property-no-vendor-prefix */
    -ms-overflow-style: scrollbar; /* Set box-sizing so that the width is not affected by padding or border */
    -ms-text-size-adjust: 100%; /* Set default font family in all browsers. */
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); /* Makes 1rem = 10px */
    -webkit-text-size-adjust: 100%; /* Prevent adjustments of font size after orientation changes in IE on Windows Phone and in iOS. */
    /* stylelint-enable property-no-vendor-prefix */
    box-sizing: border-box;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  body {
    background-color: ${({theme}) => theme.palette.background.white};

    /* Hide overflow content */
    &.overflow {
      overflow: hidden;
    }

    /* Prevent focus outline on elements that can't be accessed via keyboard. */
    [tabindex="-1"]:focus {
      outline: none !important;
    }

    /* Hide overflow in IE */
    svg:not(:root) {
      overflow: hidden;
    }

    /* Hide elements has 'hidden' attribute in IE 10- */
    [hidden] {
      display: none !important;
    }

    ::-webkit-file-upload-button {
      /* stylelint-disable-next-line property-no-vendor-prefix */
      -webkit-appearance: button;
      font: inherit;
    }
  }
`

const typographyMixin = css<StyledProps<IProps>>`
  html {
    font-family: ${({theme}) =>
      theme.typography
        .fontFamily}; /* Prevent scrollbars to overlap content in IE11 and Edge */
    font-size: ${({theme}) =>
      toPercent(
        getValue(theme.typography.htmlFontSize),
        16
      )}; /* Set default tap highlight to be transparent in iOS. */
    font-style: normal;
    color: ${({theme}) => theme.palette.text.dark};
  }

  body {
    ${bodyMixin<IProps>()}
  }

  h1 {
    ${h1Mixin<IProps>()}
  }

  h2 {
    ${h2Mixin<IProps>()}
  }

  h3 {
    ${h3Mixin<IProps>()}
  }

  h4 {
    ${h4Mixin<IProps>()}
  }

  h5 {
    ${h5Mixin<IProps>()}
  }

  h6 {
    ${h6Mixin<IProps>()}
  }

  small {
    ${smallMixin<IProps>()}
  }

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.bold};

    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    caption {
      font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
    }
  }
`

const utilsMixin = css<StyledProps<IProps>>`
  .hidden {
    visibility: hidden;
  }

  .d-none {
    display: none;
  }

  .hidden-sm-down {
    ${({theme}) =>
      mediaBreakpointDown(
        "sm",
        theme.grid.breakpoints,
        `
      display: none;
    `
      )}
  }

  .text-center {
    text-align: center;
  }
`

const CssBaseline: GlobalStyleComponent<
  IProps,
  DefaultTheme
> = createGlobalStyle`
  ${genericResetMixin}

  ${appResetMixin}

  ${typographyMixin}

  ${utilsMixin}
`

export default CssBaseline
