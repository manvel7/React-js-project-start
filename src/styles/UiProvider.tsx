import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
