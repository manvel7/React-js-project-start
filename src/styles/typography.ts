import {
  css,
  DefaultTheme,
  FlattenInterpolation,
  StyledProps,
  ThemedStyledProps
} from "styled-components"
import {pxToRem} from "../utils/unit"

const h1Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h1.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h1.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h1.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h1.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h2Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h2.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h2.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h2.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h2.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h3Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h3.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h3.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h3.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h3.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h4Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h4.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h4.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h4.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h4.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h5Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h5.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h5.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h5.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h5.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h6Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h6.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h6.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h6.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h6.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const bodyMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.body.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.body.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.lineHeight,
      theme.typography.htmlFontSize
    )};
`

const smallMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.caption.weight};
  letter-spacing: ${({theme}) =>
    theme.typography.variants.caption.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`
export {
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin,
  bodyMixin,
  smallMixin
}
