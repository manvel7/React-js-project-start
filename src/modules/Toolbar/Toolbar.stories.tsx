import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import {Dropdown, DropdownMenuItem} from "../../components/Dropdown"
import Toolbar from "./Toolbar"

const meta: Meta = {
  title: "Modules/Toolbar",
  component: Toolbar
}

const ToolbarStory: Story = (args) => (
  <Toolbar {...args}>
    <Dropdown size="sm" title="Share">
      <DropdownMenuItem>Item 1</DropdownMenuItem>
      <DropdownMenuItem>Item 2</DropdownMenuItem>
    </Dropdown>
    <Dropdown size="sm" title="Export">
      <DropdownMenuItem>Item 1</DropdownMenuItem>
      <DropdownMenuItem>Item 2</DropdownMenuItem>
    </Dropdown>
  </Toolbar>
)

const Default = ToolbarStory.bind({})

export {Default}

export default meta
