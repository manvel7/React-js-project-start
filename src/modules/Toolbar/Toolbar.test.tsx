import React from "react"
import {render} from "@testing-library/react"
import {Dropdown, DropdownMenuItem} from "../../components/Dropdown"
import {UiProvider} from "../../styles"
import Toolbar from "./Toolbar"

describe("Toolbar", () => {
  describe("Default", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Toolbar>
            <Dropdown size="sm" title="Share">
              <DropdownMenuItem>Item 1</DropdownMenuItem>
              <DropdownMenuItem>Item 2</DropdownMenuItem>
            </Dropdown>
            <Dropdown size="sm" title="Export">
              <DropdownMenuItem>Item 1</DropdownMenuItem>
              <DropdownMenuItem>Item 2</DropdownMenuItem>
            </Dropdown>
          </Toolbar>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
