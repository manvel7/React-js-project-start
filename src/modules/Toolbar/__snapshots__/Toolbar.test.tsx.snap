// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Default should match snapshot 1`] = `
.c3 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.3rem;
  padding: 0.2rem 1.6rem;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c3 b,
.c3 strong {
  font-weight: 800;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
}

.c2 button:first-child {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c2 button:last-child {
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
}

.c2 button:not(:first-child):not(:last-child) {
  border-radius: 0;
}

.c2 button.toggle__button--inactive {
  border-color: #E6E6E6;
}

.c2 button.toggle__button--active {
  font-weight: 800;
}

.c4 {
  display: none;
  min-width: 100%;
  opacity: 0;
  position: absolute;
  -webkit-transform: rotateX(-30deg);
  -ms-transform: rotateX(-30deg);
  transform: rotateX(-30deg);
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-transition: opacity 0.15s ease-in,-webkit-transform 0.15s ease-in;
  -webkit-transition: opacity 0.15s ease-in,transform 0.15s ease-in;
  transition: opacity 0.15s ease-in,transform 0.15s ease-in;
  z-index: 5;
}

.c1 {
  display: inline-block;
  -webkit-perspective: 30rem;
  -moz-perspective: 30rem;
  -ms-perspective: 30rem;
  perspective: 30rem;
  position: relative;
  z-index: 3;
}

.c5 {
  background-color: #F7F7F7;
  border: 0.1rem solid #E6E6E6;
  border-radius: 0.4rem;
  cursor: pointer;
  display: block;
  padding: 0.8rem 1.6rem;
  -webkit-transition: background-color 0.15s ease-in;
  transition: background-color 0.15s ease-in;
  width: 100%;
}

.c5:hover {
  background-color: #F1F2F5;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 0.1rem solid #E6E6E6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.55rem;
}

.c0 > div > div {
  margin-right: 1.6rem;
}

.c0 .return-button {
  margin-right: 1.6rem;
}

@media (min-width:576px) {
  .c0 .return-button {
    visibility: hidden;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <button
        class="c3"
      >
        Share
      </button>
      <button
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-caret-down fa-w-10 fa-sm "
          data-icon="caret-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
      >
        Item 1
      </a>
      <a
        class="c5"
      >
        Item 2
      </a>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <button
        class="c3"
      >
        Export
      </button>
      <button
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-caret-down fa-w-10 fa-sm "
          data-icon="caret-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
      >
        Item 1
      </a>
      <a
        class="c5"
      >
        Item 2
      </a>
    </div>
  </div>
</div>
`;
